### DISCUSSION TITLE
Integration doesn't work since update HA 2023.6.3

### DISCUSSION THREAD

**Discussion by MichT368** [Hyundai-Kia-Connect/kia_uvo#671] [Q&A]
The integration doesn't work anymore since update HA 2023.6.3
Is there anybody else having this?

My error log says the following:

>  File "/config/custom_components/kia_uvo/coordinator.py", line 107, in _async_update_data
>     await self.async_check_and_refresh_token()
>   File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
>     await self.hass.async_add_executor_job(
>   File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
>     result = self.fn(*self.args, **self.kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 118, in check_and_refresh_token
>     self.initialize()
>   File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 62, in initialize
>     self.token: Token = self.api.login(self.username, self.password)
>                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
>     device_id = self._get_device_id(stamp)
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
>     _check_response_for_errors(response)
>   File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
>     raise APIError(f"Server returned: '{response['resMsg']}'")
> hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Invalid parameter.'
> 2023-07-04 07:55:35.740 DEBUG (MainThread) [custom_components.kia_uvo.coordinator] Finished fetching kia_uvo data in 0.263 seconds (success: False)

I haven't changed anything, nor am I aware that any of my account settings might have changed.
Any ideas?
Thank you in advance.
Michael

**Comment by cdnninja** [comment]
Did you update the integration yet?On Jul 4, 2023, at 12:13 AM, MichT368 ***@***.***> wrote:﻿
The integration doesn't work anymore since update HA 2023.6.3
Is there anybody else having this?
My error log says the following:

File "/config/custom_components/kia_uvo/coordinator.py", line 107, in _async_update_data
await self.async_check_and_refresh_token()
File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
await self.hass.async_add_executor_job(
File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
result = self.fn(*self.args, **self.kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 118, in check_and_refresh_token
self.initialize()
File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 62, in initialize
self.token: Token = self.api.login(self.username, self.password)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
device_id = self._get_device_id(stamp)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
_check_response_for_errors(response)
File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Invalid parameter.'
2023-07-04 07:55:35.740 DEBUG (MainThread) [custom_components.kia_uvo.coordinator] Finished fetching kia_uvo data in 0.263 seconds (success: False)

I haven't changed anything, nor am I aware that any of my account settings might have changed.
Any ideas?
Thank you in advance.
Michael

**Reply by MichT368** [reply]
Hard to believe that I was so stupid. Since a while, HACS has its own update section. I wasn't aware of that. Thank you very much, that solved the problem.
Michael
